scripts/update-component-sha.sh

```diff
@@ -1,10 +1,66 @@
 #!/bin/sh
 set -e
-if [ $# -ne 2 ] ; then
-    echo "Need <OLD> and <NEW> as arguments" >&2
+
+##
+#
+# script to replace hashes in config files
+# can be run in 2 modes:
+# 1- $0 --hash $oldhash $newhash              - replace all instances of $oldhash with $newhash
+# 2- $0 --image $image $newhash        - replace all hashes for $image with $newhash
+#
+# for backwards compatibility if no argtype is given, the first method is assumed. Thus the following are equivalent:
```

Comment:
It doesn't look like this is what the code does. Frankly for a tool this new there is no need to maintain backwards compatibility, so just remove this comment, but also see below